src/Sentry/MetricHelper.cs

```diff
@@ -44,28 +44,47 @@ internal static long GetTimeBucketKey(this DateTimeOffset timestamp)
         .Subtract(TimeSpan.FromMilliseconds(FlushShift));
 
 #if NET7_0_OR_GREATER
-    [GeneratedRegex(InvalidKeyCharactersPattern, RegexOptions.Compiled)]
-    private static partial Regex InvalidKeyCharacters();
-    internal static string SanitizeKey(string input) => InvalidKeyCharacters().Replace(input, "_");
+    [GeneratedRegex(InvalidMetricKeyOrNameCharactersPattern, RegexOptions.Compiled)]
+    private static partial Regex InvalidMetricKeyOrNameCharacters();
+    internal static string SanitizeMetricKeyOrName(string input) => InvalidMetricKeyOrNameCharacters().Replace(input, "_");
 
-    [GeneratedRegex(InvalidValueCharactersPattern, RegexOptions.Compiled)]
-    private static partial Regex InvalidValueCharacters();
-    internal static string SanitizeValue(string input) => InvalidValueCharacters().Replace(input, "");
+    [GeneratedRegex(InvalidTagKeyCharactersPattern, RegexOptions.Compiled)]
+    private static partial Regex InvalidTagKeyCharacters();
+    internal static string SanitizeTagKey(string input) => InvalidTagKeyCharacters().Replace(input, "");
 
     [GeneratedRegex(InvalidMetricUnitCharactersPattern, RegexOptions.Compiled)]
     private static partial Regex InvalidMetricUnitCharacters();
-    internal static string SanitizeMetricUnit(string input) => InvalidMetricUnitCharacters().Replace(input, "_");
+    internal static string SanitizeMetricUnit(string input) => InvalidMetricUnitCharacters().Replace(input, "");
 #else
-    private static readonly Regex InvalidKeyCharacters = new(InvalidKeyCharactersPattern, RegexOptions.Compiled);
-    internal static string SanitizeKey(string input) => InvalidKeyCharacters.Replace(input, "_");
+    private static readonly Regex InvalidMetricKeyOrNameCharacters = new(InvalidMetricKeyOrNameCharactersPattern, RegexOptions.Compiled);
+    internal static string SanitizeMetricKeyOrName(string input) => InvalidMetricKeyOrNameCharacters.Replace(input, "_");
```

Comment:
Just double checking: We're keeping the `_` for MetricKeyOrName only?

Reply:
Yeah, that's per [the new docs](https://develop.sentry.dev/sdk/metrics/#normalization)... slight difference between the allowed characters for each different key or value type as well (e.g. tag keys allow a `/` but metric keys don't).